src/main/java/org/wso2/carbon/esb/connector/NatsConnection.java

```diff
@@ -0,0 +1,253 @@
+    }
+
+    /**
+     * Check whether the parameter provided is empty or not.
+     *
+     * @param serverConfig the server configuration properties object.
+     * @param property the property to set.
+     * @param parameter the parameter value to set to the property.
+     */
+    private void setServerConfigProperty(Properties serverConfig, String property, String parameter) {
+        if (StringUtils.isNotEmpty(parameter)) {
+            serverConfig.setProperty(property, parameter);
+        }
+    }
+
+    /**
+     * Create the SSLContext to establish connection with TLS.
+     *
+     * @param tlsConnection the TLS connection object.
+     * @return the SSLContext or null if any exceptions.
+     */
+    private static SSLContext createSSLContext(TLSConnection tlsConnection) {
+        Log log = LogFactory.getLog(NatsConnection.class);
+        try {
+            KeyStore keyStore = loadKeyStore(tlsConnection.getKeyStoreType(), tlsConnection.getKeyStoreLocation(), tlsConnection.getTrustStorePassword());
+            KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(tlsConnection.getKeyManagerAlgorithm().equals("") ? NatsConstants.DEFAULT_TLS_ALGORITHM : tlsConnection.getKeyManagerAlgorithm());
+            keyManagerFactory.init(keyStore, tlsConnection.getKeyStorePassword().toCharArray());
+
+            KeyStore trustStore = loadKeyStore(tlsConnection.getTrustStoreType(), tlsConnection.getTrustStoreLocation(), tlsConnection.getTrustStorePassword());
+            TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(tlsConnection.getTrustManagerAlgorithm().equals("") ? NatsConstants.DEFAULT_TLS_ALGORITHM : tlsConnection.getTrustManagerAlgorithm());
+            trustManagerFactory.init(trustStore);
+
+            SSLContext sslContext = SSLContext.getInstance(tlsConnection.getProtocol().equals("") ? Options.DEFAULT_SSL_PROTOCOL : tlsConnection.getProtocol());
+            sslContext.init(keyManagerFactory.getKeyManagers(), trustManagerFactory.getTrustManagers(), new SecureRandom());
+            return sslContext;
+        } catch (KeyStoreException | IOException | CertificateException | NoSuchAlgorithmException | UnrecoverableKeyException | KeyManagementException e) {
+            log.error("Invalid TLS parameters. Establishing connection without TLS if possible.", e);
+            return null;
+        }
+    }
+
+    /**
+     * Load key store and trust store.
+     *
+     * @param storeType the type of store file.
+     * @param storeLocation the location of store file.
+     * @param trustStorePassword the password of trust store file.
+     * @return the store.
+     */
+    private static KeyStore loadKeyStore(String storeType, String storeLocation, String trustStorePassword) throws KeyStoreException, IOException, CertificateException, NoSuchAlgorithmException {
+        KeyStore store = KeyStore.getInstance(storeType.equals("") ? NatsConstants.DEFAULT_STORE_TYPE : storeType);
```

Comment:
Can storeType be null? If so, better to switch "" and storeType to avoid NPE

Reply:
If storeType is null it is converted to an empty string in another class.